Kernel panic during normal use

I just listened to music and browsed. Looks strongly zfs correlated but not familiar:

````
Anonymous UUID:       67548459-A116-8E77-7ACF-DCDFC462FC4B

Sat Jan  2 18:01:19 2016

*** Panic Report ***
panic(cpu 0 caller 0xffffff8019fd201a): "Double fault at 0xffffff8019f1b5b4, registers:\n" "CR0: 0x0000000080010033, CR2: 0xffffff91300bfff8, CR3: 0x00000000501ac000, CR4: 0x0000000000002660\n" "RAX: 0xffffff91300c007c, RBX: 0xffffff80fac25cf0, RCX: 0xffffff91300c00b4, RDX: 0xffffff91300c00f8\n" "RSP: 0xffffff91300c0000, RBP: 0xffffff91300c0000, RSI: 0xffffff913079b628, RDI: 0xffffff91300c0028\n" "R8:  0x00000000ffffffff, R9:  0x0000000000000002, R10: 0x0000000000000007, R11: 0xffffff8043fdb5c8\n" "R12: 0xffffff91300c0018, R13: 0x0000000000000002, R14: 0x0000000000000000, R15: 0x00000000ffffffff\n" "RFL: 0x0000000000010082, RIP: 0xffffff8019f1b5b4, CS:  0x0000000000000008, SS:  0x0000000000000010\n" "Error code: 0x0000000000000000\n"@/Library/Caches/com.apple.xbs/Sources/xnu/xnu-3248.20.55/osfmk/i386/trap_native.c:168
Backtrace (CPU 0), Frame : Return Address
0xffffff80fb537cf0 : 0xffffff8019ede792 mach_kernel : _panic + 0xe2
0xffffff80fb537d70 : 0xffffff8019fd201a mach_kernel : _panic_64 + 0x1ea
0xffffff80fb537ed0 : 0xffffff8019fefaef mach_kernel : _hndl_double_fault + 0xf
0xffffff91300c0000 : 0xffffff8019f1bb4b mach_kernel : _waitq_wakeup64_thread + 0xe5b
0xffffff91300c00a0 : 0xffffff8019f18b89 mach_kernel : _waitq_wakeup64_one_locked + 0x29
0xffffff91300c00e0 : 0xffffff8019f1abc0 mach_kernel : _waitq_wakeup64_one + 0x70
0xffffff91300c0120 : 0xffffff8019ef375d mach_kernel : _thread_wakeup_prim + 0x8d
0xffffff91300c0160 : 0xffffff7f9aa93513 net.lundman.zfs : _zil_commit + 0x9f7
0xffffff91300c0280 : 0xffffff7f9aa81fbf net.lundman.zfs : _zfs_fsync + 0xf1
0xffffff91300c02b0 : 0xffffff801a101cc7 mach_kernel : _vfs_addtrigger + 0x1ba7
0xffffff91300c0330 : 0xffffff801a101976 mach_kernel : _vfs_addtrigger + 0x1856
0xffffff91300c0370 : 0xffffff801a104f24 mach_kernel : _vfs_addtrigger + 0x4e04
0xffffff91300c03a0 : 0xffffff801a100a21 mach_kernel : _vfs_addtrigger + 0x901
0xffffff91300c0430 : 0xffffff7f9aa8a805 net.lundman.zfs : _zfs_znode_getvnode + 0x1a3
0xffffff91300c04d0 : 0xffffff7f9aa8f9b0 net.lundman.zfs : _zfs_zget_ext + 0x356
0xffffff91300c0570 : 0xffffff7f9aa6a82d net.lundman.zfs : _zfs_rmnode + 0x3a3
0xffffff91300c0770 : 0xffffff7f9aa88da8 net.lundman.zfs : _zfs_vnop_reclaim + 0x116
0xffffff91300c07b0 : 0xffffff801a101e4a mach_kernel : _vfs_addtrigger + 0x1d2a
0xffffff91300c0830 : 0xffffff801a101976 mach_kernel : _vfs_addtrigger + 0x1856
0xffffff91300c0870 : 0xffffff801a0fb5f9 mach_kernel : _vnode_put_locked + 0xb9
0xffffff91300c08a0 : 0xffffff801a0e726b mach_kernel : _vnode_update_identity + 0x3db
0xffffff91300c0900 : 0xffffff801a101e81 mach_kernel : _vfs_addtrigger + 0x1d61
0xffffff91300c0980 : 0xffffff801a101976 mach_kernel : _vfs_addtrigger + 0x1856
0xffffff91300c09c0 : 0xffffff801a104f24 mach_kernel : _vfs_addtrigger + 0x4e04
0xffffff91300c09f0 : 0xffffff801a100a21 mach_kernel : _vfs_addtrigger + 0x901
0xffffff91300c0a80 : 0xffffff7f9aa8a805 net.lundman.zfs : _zfs_znode_getvnode + 0x1a3
0xffffff91300c0b20 : 0xffffff7f9aa8f9b0 net.lundman.zfs : _zfs_zget_ext + 0x356
0xffffff91300c0bc0 : 0xffffff7f9aa6a82d net.lundman.zfs : _zfs_rmnode + 0x3a3
0xffffff91300c0dc0 : 0xffffff7f9aa88da8 net.lundman.zfs : _zfs_vnop_reclaim + 0x116
0xffffff91300c0e00 : 0xffffff801a101e4a mach_kernel : _vfs_addtrigger + 0x1d2a
0xffffff91300c0e80 : 0xffffff801a101976 mach_kernel : _vfs_addtrigger + 0x1856
0xffffff91300c0ec0 : 0xffffff801a0fb5f9 mach_kernel : _vnode_put_locked + 0xb9
0xffffff91300c0ef0 : 0xffffff801a0e726b mach_kernel : _vnode_update_identity + 0x3db
0xffffff91300c0f50 : 0xffffff801a101e81 mach_kernel : _vfs_addtrigger + 0x1d61
0xffffff91300c0fd0 : 0xffffff801a101976 mach_kernel : _vfs_addtrigger + 0x1856
0xffffff91300c1010 : 0xffffff801a104f24 mach_kernel : _vfs_addtrigger + 0x4e04
0xffffff91300c1040 : 0xffffff801a100a21 mach_kernel : _vfs_addtrigger + 0x901
0xffffff91300c10d0 : 0xffffff7f9aa8a805 net.lundman.zfs : _zfs_znode_getvnode + 0x1a3
0xffffff91300c1170 : 0xffffff7f9aa8f9b0 net.lundman.zfs : _zfs_zget_ext + 0x356
0xffffff91300c1210 : 0xffffff7f9aa6a82d net.lundman.zfs : _zfs_rmnode + 0x3a3
0xffffff91300c1410 : 0xffffff7f9aa88da8 net.lundman.zfs : _zfs_vnop_reclaim + 0x116
0xffffff91300c1450 : 0xffffff801a101e4a mach_kernel : _vfs_addtrigger + 0x1d2a
0xffffff91300c14d0 : 0xffffff801a101976 mach_kernel : _vfs_addtrigger + 0x1856
0xffffff91300c1510 : 0xffffff801a0fb5f9 mach_kernel : _vnode_put_locked + 0xb9
0xffffff91300c1540 : 0xffffff801a0e726b mach_kernel : _vnode_update_identity + 0x3db
0xffffff91300c15a0 : 0xffffff801a101e81 mach_kernel : _vfs_addtrigger + 0x1d61
0xffffff91300c1620 : 0xffffff801a101976 mach_kernel : _vfs_addtrigger + 0x1856
0xffffff91300c1660 : 0xffffff801a104f24 mach_kernel : _vfs_addtrigger + 0x4e04
0xffffff91300c1690 : 0xffffff801a100a21 mach_kernel : _vfs_addtrigger + 0x901
0xffffff91300c1720 : 0xffffff7f9aa8a805 net.lundman.zfs : _zfs_znode_getvnode + 0x1a3
0xffffff91300c17c0 : 0xffffff7f9aa8f9b0 net.lundman.zfs : _zfs_zget_ext + 0x356
0xffffff91300c1860 : 0xffffff7f9aa6a82d net.lundman.zfs : _zfs_rmnode + 0x3a3
0xffffff91300c1a60 : 0xffffff7f9aa88da8 net.lundman.zfs : _zfs_vnop_reclaim + 0x116
0xffffff91300c1aa0 : 0xffffff801a101e4a mach_kernel : _vfs_addtrigger + 0x1d2a
0xffffff91300c1b20 : 0xffffff801a101976 mach_kernel : _vfs_addtrigger + 0x1856
0xffffff91300c1b60 : 0xffffff801a0fb5f9 mach_kernel : _vnode_put_locked + 0xb9
0xffffff91300c1b90 : 0xffffff801a0e726b mach_kernel : _vnode_update_identity + 0x3db
0xffffff91300c1bf0 : 0xffffff801a101e81 mach_kernel : _vfs_addtrigger + 0x1d61
0xffffff91300c1c70 : 0xffffff801a101976 mach_kernel : _vfs_addtrigger + 0x1856
0xffffff91300c1cb0 : 0xffffff801a104f24 mach_kernel : _vfs_addtrigger + 0x4e04
0xffffff91300c1ce0 : 0xffffff801a100a21 mach_kernel : _vfs_addtrigger + 0x901
0xffffff91300c1d70 : 0xffffff7f9aa8a805 net.lundman.zfs : _zfs_znode_getvnode + 0x1a3
0xffffff91300c1e10 : 0xffffff7f9aa8f9b0 net.lundman.zfs : _zfs_zget_ext + 0x356
0xffffff91300c1eb0 : 0xffffff7f9aa6a82d net.lundman.zfs : _zfs_rmnode + 0x3a3
0xffffff91300c20b0 : 0xffffff7f9aa88da8 net.lundman.zfs : _zfs_vnop_reclaim + 0x116
0xffffff91300c20f0 : 0xffffff801a101e4a mach_kernel : _vfs_addtrigger + 0x1d2a
0xffffff91300c2170 : 0xffffff801a101976 mach_kernel : _vfs_addtrigger + 0x1856
0xffffff91300c21b0 : 0xffffff801a0fb5f9 mach_kernel : _vnode_put_locked + 0xb9
0xffffff91300c21e0 : 0xffffff801a0e726b mach_kernel : _vnode_update_identity + 0x3db
0xffffff91300c2240 : 0xffffff801a101e81 mach_kernel : _vfs_addtrigger + 0x1d61
0xffffff91300c22c0 : 0xffffff801a101976 mach_kernel : _vfs_addtrigger + 0x1856
0xffffff91300c2300 : 0xffffff801a104f24 mach_kernel : _vfs_addtrigger + 0x4e04
0xffffff91300c2330 : 0xffffff801a100a21 mach_kernel : _vfs_addtrigger + 0x901
0xffffff91300c23c0 : 0xffffff7f9aa8a805 net.lundman.zfs : _zfs_znode_getvnode + 0x1a3
0xffffff91300c2460 : 0xffffff7f9aa8f9b0 net.lundman.zfs : _zfs_zget_ext + 0x356
0xffffff91300c2500 : 0xffffff7f9aa6a82d net.lundman.zfs : _zfs_rmnode + 0x3a3
0xffffff91300c2700 : 0xffffff7f9aa88da8 net.lundman.zfs : _zfs_vnop_reclaim + 0x116
0xffffff91300c2740 : 0xffffff801a101e4a mach_kernel : _vfs_addtrigger + 0x1d2a
0xffffff91300c27c0 : 0xffffff801a101976 mach_kernel : _vfs_addtrigger + 0x1856
0xffffff91300c2800 : 0xffffff801a0fb5f9 mach_kernel : _vnode_put_locked + 0xb9
	Backtrace continues...
      Kernel Extensions in backtrace:
         net.lundman.zfs(1.5)[C1866444-37BF-34B9-9A6A-A8331C0E8752]@0xffffff7f9a9fb000->0xffffff7f9ac3bfff
            dependency: com.apple.iokit.IOStorageFamily(2.1)[956BA336-5C0A-37E9-B395-538D1BD95AC0]@0xffffff7f9a64c000
            dependency: net.lundman.spl(1.5.0)[13BE3A00-D6A8-3F25-A9C6-D2779281BAA1]@0xffffff7f9a9ad000

BSD process name corresponding to current thread: arqcommitter
Boot args: keepsyms=1 -s

Mac OS version:
15C50

Kernel version:
Darwin Kernel Version 15.2.0: Fri Nov 13 19:56:56 PST 2015; root:xnu-3248.20.55~2/RELEASE_X86_64
Kernel UUID: 17EA3101-D2E4-31BF-BDA9-931F51049F93
Kernel slide:     0x0000000019c00000
Kernel text base: 0xffffff8019e00000
__HIB  text base: 0xffffff8019d00000
System model name: MacBookPro5,5 (Mac-F2268AC8)

System uptime in nanoseconds: 50628472790422
last loaded kext at 27200102865760: com.apple.filesystems.smbfs	3.0.0 (addr 0xffffff7f9cb65000, size 409600)
last unloaded kext at 26917744844019: com.apple.driver.usb.IOUSBHostHIDDevice	1.0.1 (addr 0xffffff7f9cb65000, size 40960)
loaded kexts:
org.virtualbox.kext.VBoxNetAdp	5.0.10
org.virtualbox.kext.VBoxNetFlt	5.0.10
org.virtualbox.kext.VBoxUSB	5.0.10
org.virtualbox.kext.VBoxDrv	5.0.10
net.lundman.zfs	1.5.0
net.lundman.spl	1.5.0
com.apple.filesystems.smbfs	3.0.0
com.apple.driver.AppleBluetoothMultitouch	90.3
com.apple.driver.AudioAUUC	1.70
com.apple.driver.AppleHWSensor	1.9.5d0
com.apple.iokit.IOBluetoothSerialManager	4.4.3f4
com.apple.driver.AGPM	110.20.21
com.apple.filesystems.autofs	3.0
com.apple.driver.AppleOSXWatchdog	1
com.apple.driver.AppleMikeyHIDDriver	124
com.apple.driver.AppleHDA	272.51.3
com.apple.driver.AppleMik


 
````

it doesn't compile right now:

````
zfs_acl.c:2490:30: warning: incompatible pointer types passing 'cred_t *' (aka 'struct opaque_cred_t *') to parameter of type 'kauth_cred_t' (aka 'struct ucred *') [-Wincompatible-pointer-types]
        if (groupmember(zdp->z_gid, cr)) {
                                    ^~
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk/System/Library/Frameworks/Kernel.framework/Headers/sys/kauth.h:264:49: note: passing argument to parameter 'cred' here
extern int      groupmember(gid_t gid, kauth_cred_t cred);
                                                    ^
  CC       zfs-zfs_ctldir.o
  CC       zfs-zfs_debug.o
4 warnings generated.
  CC       zfs-zfs_dir.o
  CC       zfs-zfs_fm.o
zfs_dir.c:781:1: error: function definition is not allowed here
{
^
zfs_dir.c:794:1: error: function definition is not allowed here
{
^
zfs_dir.c:891:1: error: function definition is not allowed here
{
^
zfs_dir.c:922:1: error: function definition is not allowed here
{
^
zfs_dir.c:1039:1: error: function definition is not allowed here
{
^
zfs_dir.c:1045:1: error: function definition is not allowed here
{
^
zfs_dir.c:1134:1: error: function definition is not allowed here
{
^
zfs_dir.c:1209:1: error: function definition is not allowed here
{
^
zfs_dir.c:1230:2: error: expected '}'
}
 ^
zfs_dir.c:667:1: note: to match this '{'
{
^
9 errors generated.
make[3]: *** [zfs-zfs_dir.o] Error 1
make[3]: *** Waiting for unfinished jobs....
make[2]: *** [all-recursive] Error 1
make[1]: *** [all-recursive] Error 1
make: *** [all] Error 2

````

--- a/module/zfs/zfs_dir.c
+++ b/module/zfs/zfs_dir.c
@@ -714,7 +714,8 @@ zfs_rmnode(znode_t *zp)
 	error = sa_lookup(zp->z_sa_hdl, SA_ZPL_XATTR(zfsvfs),
 		&xattr_obj, sizeof (xattr_obj));
 	if (error == 0 && xattr_obj) {
-		error = zfs_zget(zfsvfs, xattr_obj, &xzp);
+		error = zfs_zget_ext(zfsvfs, xattr_obj, &xzp,
+		    ZGET_FLAG_WITHOUT_VNODE);
 		ASSERT(error == 0);
 	}
 
@@ -766,8 +767,14 @@ zfs_rmnode(znode_t *zp)
 
 	dmu_tx_commit(tx);
 out:
-	if (xzp)
-		VN_RELE_ASYNC(ZTOV(xzp), dsl_pool_vnrele_taskq(dmu_objset_pool(zfsvfs->z_os)));
+	if (xzp) {
+		/* Only release object if we are the only user */
+		if (ZTOV(xzp) == NULL)
+			zfs_zinactive(xzp);
+		else
+			VN_RELE_ASYNC(ZTOV(xzp), dsl_pool_vnrele_taskq(
+			    dmu_objset_pool(zfsvfs->z_os)));
+	}
 }
 
 static uint64_t

